Source/ARTDeviceDetails.h

```diff
@@ -40,6 +40,11 @@ NS_ASSUME_NONNULL_BEGIN
  */
 @property (strong, nonatomic) ARTDevicePushDetails *push;
 
+/**
+ * A unique device secret generated by the Ably SDK.
+ */
+@property (strong, nullable, nonatomic) ARTDeviceSecret *secret;
```

Comment:
Good points thanks, I had done this quick change as `DeviceDetails` was being used in transport and tests, not `LocalDevice` as version 2.0 api wouldn't let this being used without a `deviceSecret`.  Now I changed types in both transport and test  efb1b40e52b466fa41211c789527c1a560826392. I would assume that should be fine as DeviceDetails is a supertype but still I am not sure whether this change would break another spec point